service/history/shard/controller_impl.go

```diff
@@ -360,8 +361,11 @@ func (c *ControllerImpl) acquireShards() {
 	}
 
 	// Submit tasks to the channel.
+	numShards := c.config.NumberOfShards
+	randomStartOffset := rand.Int31n(numShards)
```

Comment:
there's only a hotspot if ringpop gets confused and says that two nodes own the same shard, right?